docker-compose.yml.example

```diff
@@ -60,6 +60,8 @@ services:
       - 8080:8080
     env_file:
       - ./config/cadence.env
+    environment:
+        - CSERVER_LOGLEVEL=$LOG_LEVEL
```

Comment:
I'm open to changing this mechanism, of course - I was looking for something that can be done when running the server, and is simple to be communicated when debugging an issue (and could theoretically be reused if we had more containers, at that), and causes no errors if omitted.

The last part isn't 100% there - It doesn't break functionality whatsoever if you omit the level (you just get Info, as before), but you do get this mildly annoying warning when you run compose:
![image](https://github.com/user-attachments/assets/a548e0c9-45e4-4a72-baa8-486f4e565f8e)

Reply:
I like this PR and think a log level option set is definitely something I overlooked. I do see the value in usage/communication that an inline method suggested would provide. That said, I think for simplicity, we should consider keeping the way it gets set through `cadence.env` to stay consolidated and to keep the warning you mentioned suppressed. The warning might confuse a beginner and send them asking how it can be set.

Further, we could default that option to `CSERVER_DEVMODE=debug` and all cadence users would just have sharable debugging logs by default (without needing to set a level (without seeing a warning)). This decreases complexity for any user reporting issues who can just one-step copy paste their logs without us asking them to reproduce with debug on. Further, it makes the quieting of logs an explicit/deliberate action reserved for more advanced users only.

I'm looking specifically at the "development" section of the config:

https://github.com/kenellorando/cadence/blob/a287dcf9672a729460e2cb682baa8dd820e67c28/config/cadence.env.example#L9-L12
where we could do
```
CSERVER_LOG_LEVEL=debug # acceptable values = debug (default), info, warn, error
```

Reply:
> I like this PR and think a log level option set is definitely something I overlooked. I do see the value in usage/communication that an inline method suggested would provide. That said, I think for simplicity, we should consider keeping the way it gets set through `cadence.env` to stay consolidated and to keep the warning you mentioned suppressed. The warning might confuse a beginner and send them asking how it can be set.

I dislike the warning as well.

The reason I wanted to keep it out of cadence.env was to facilitate users being able to set the log level without doing a reinstall (if we provide install to provide configuration for them, then the assumption is that they won't be touching their .env files) - Thus ensuring that if an issue comes in and we ask for them to run it again with the debug level enabled, that occurs with the same configuration (we won't have install bugs that fix themselves by rerunning install as a diagnostic step).

> Further, we could default that option to `CSERVER_DEVMODE=debug` and all cadence users would just have sharable debugging logs by default (without needing to set a level (without seeing a warning)). This decreases complexity for any user reporting issues who can just one-step copy paste their logs without us asking them to reproduce with debug on. Further, it makes the quieting of logs an explicit/deliberate action reserved for more advanced users only.

That is also fair - Should we want to state that debug happens all the time and controlling logs is for advanced users.

That's not to say that I'm strongly opposed to that idea - Only that I want us to be on the same page about how we're describing this setting (the user is not allowed to control what they see unless they're knowledgeable about the internals of Cadence vs we suggest a default to the user on install and they can set it as they like)

Reply:
... Actually, as it turns out, `docker compose` supports the bash syntax to provide a default (`${LOG_LEVEL:-debug}`) - Doing that suppresses the warning at least, so if you like my reasoning to have it this way it'll now work without producing the warning. So I guess that leaves us with:

1. Do what I did originally, except with no warning (user can reset log level without reinstall or manual reconfiguration)
2. Do as you've suggested (user cannot reset log level except through manual reconfiguration)
3. Add it as a sixth option to the install (user can reset the log level, but must reinstall to do so)

The real core of which is best depends on how easy we want it to be to change the log level - I think it's best to pick either 1 or 2, and either require that users produce all the logs we expect to need from them or make it nearly trivial for them to change the log level on demand.

Reply:
I think we should keep this contained to `cadence.env`. The install script doesn't touch all values in `cadence.env` as you know-- only some with the highest probability that would need to be changed. Others, such as `CSERVER_DEVMODE`, are only accessible by manual manipulation of the file. I can change any value in the config without necessarily "installing" and a `docker compose up` would pass the new values all the same. A `CSERVER_LOG_LEVEL` value makes sense to me in the "Development" section (rarely changed enough that it wouldn't require an option in the install script).

But, to get more to the main point, core to me here is the preference to keep handling configuration as consolidated as possible, rather than open another channel that operates outside of the bounds we currently have. I appreciate the thought as a convenience factor, but I think it makes sense to keep them together for simplicity.

Reply:
Alright - If you want to conceptualize this as something that's set only by devs or with dev instruction, that's fine by me (I expect users to run install if they want to configure stuff, and not to understand/be informed in the usage of config files without assistance).